### DISCUSSION TITLE
Supabase auth for same session on multiple sub domains

### DISCUSSION THREAD

**Discussion by loekj** [supabase/supabase#5742] [Questions]
Hi,
Is it possible to have a session created on app.mydomain.com be valid on *.mydomain.com?
I am hosting a simple landing page on Vercel and it will need to have a logged-in section as well; accessing the user's date to alter the landing page.
On app.mydomain.com will be a separate React app.

I'm considering using Supertokens.io because they are more flexible than Auth0 and it should be possible there. Even better I'd stick with Supabase if this is possible. Anyone here has any idea? :)

**Comment by romshiri** [comment]
Did you find any solution to this? facing the same issue..

**Reply by gitmolejon** [reply]
Same here 👀

**Reply by jtiulentino** [reply]
me too...

**Reply by jtiulentino** [reply]
According to this issue, it should be possible... but I followed the instructions in there to add wildcard subdomains, and I still get a null User object in the subdomains. 
Supabase auth currently does not allow a single login session across subdomains: https://github.com/supabase/supabase/issues/473

**Reply by NateIsern** [reply]
Sameeeee, but I thought about making a gateway using rest API only for the user, I hope it works.

**Comment by hf** [comment]
Hey everyone, I'm on the Auth team.

If I understand correctly, your question is about whether two different apps, that exist both on `a.example.com` and `b.example.com` can share the same login session?

This is definitely doable, but it's not well documented.

**When using Auth Helpers**

You can pass a `domain` option when calling `createBrowserSupabaseClient` which will follow the [semantics of the `Domain` cookie attribute](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#domain_attribute).

**When using the JS library on your own**

You can do this by using a hand-crafted cookie using the [`Domain` cookie attribute](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#domain_attribute) to share existing access and refresh tokens with the other site. 

In both of your applications, you need to add code similar to this:

```typescript
supabase.auth.onAuthStateChange((event, session) => {
  if (event === 'SIGNED_OUT' || event === 'USER_DELETED') {
    // delete cookies on sign out
    const expires = new Date(0).toUTCString()
    document.cookie = `my-access-token=; Domain=example.com; path=/; expires=${expires}; SameSite=Lax; secure`
    document.cookie = `my-refresh-token=; Domain=example.com; path=/; expires=${expires}; SameSite=Lax; secure`
  } else if (event === 'SIGNED_IN' || event === 'TOKEN_REFRESHED') {
    const maxAge = 100 * 365 * 24 * 60 * 60 // 100 years, never expires
    document.cookie = `my-access-token=${session.access_token}; Domain=example.com; path=/; max-age=${maxAge}; SameSite=Lax; secure`
    document.cookie = `my-refresh-token=${session.refresh_token}; Domain=example.com; path=/; max-age=${maxAge}; SameSite=Lax; secure`
  }
})
```

Each time a new access and refresh tokens are issued _on any site_ these cookies will get updated. Then you should also add this code next to the above callback:

```typescript
const cookies = document.cookie.split(/\s*;\s*/).map(cookie => cookie.split('='));
const accessTokenCookie = cookies.find(x => x[0] == 'my-access-token');
const refreshTokenCookie = cookies.find(x => x[0] == 'my-refresh-token');

if (accessTokenCookie && refreshTokenCookie) {
  await supabase.auth.setSession({
    access_token: accessTokenCookie[1],
    refresh_token: refreshTokenCookie[1],
  })
}
```

This snippet extracts the cookies (that either of the sites set) and forces the client library to use those.

Notice the addition of `Domain=example.com`. Per the MDN documentation:

> The Domain attribute specifies which [host](https://developer.mozilla.org/en-US/docs/Glossary/Host)s can receive a cookie. If unspecified, the attribute defaults to the same host that set the cookie, excluding subdomains. If Domain is specified, then subdomains are always included. Therefore, specifying Domain is less restrictive than omitting it. However, it can be helpful when subdomains need to share information about a user.

> For example, if you set Domain=mozilla.org, cookies are available on subdomains like developer.mozilla.org.

If you need multiple domains, you would need to set cookies multiple times with the different top-level domain.

Note that this cookie approach does not signal to tabs currently opened on the other site. This is very difficult to do right, but you can improve the user experience by checking for the cookies (as shown in the snippet above) just before you call a `signIn` function.

**Reply by NateIsern** [reply]
What about using different domains? But with a central domain for authentication (auth.domain.com)

**Reply by kernelguardian** [reply]
`createBrowserSupabaseClient` seems to be depreciated. https://supabase.com/docs/guides/auth/auth-helpers/nextjs#migrating-to-v07x

**Reply by WJimmyCook** [reply]
It seems that calling setSession() triggers a SIGNED_IN event, so if you call setSession every time a SIGNED_IN event is triggered in onAuthStateChange then it results in an infinite loop and every loop makes a request the /user endpoint in supabase. Am I doing something wrong?

**Reply by bubbleheadinc** [reply]
@WJimmyCook Same issue. Ever find a solution?

**Reply by WJimmyCook** [reply]
@bubbleheadinc nope, unfortunately never figured it out. Just ended up not implementing it. Had to move on to higher priority stuff.

**Comment by berik** [comment]
Example of using Auth helper in Next.js using TS
```
export default function App({
  Component,
  pageProps,
}: AppProps<{ initialSession: Session }>) {
  // Create a new supabase browser client on every first render.
  const [supabaseClient] = useState(() =>
    createBrowserSupabaseClient<Database>({
      cookieOptions: {
        domain: "example.com",
        maxAge: "100000000",
        path: "/",
        sameSite: "Lax",
        secure: "secure",
      },
    })
  );

  return (
    <SessionContextProvider
      supabaseClient={supabaseClient}
      initialSession={pageProps.initialSession}
    >
      <Component {...pageProps} />
    </SessionContextProvider>
  );
}
```

if authenticate on main page, then the cookie will be available on `a.example.com` and `b.example.com`

**Reply by chhuang** [reply]
I tried on with `.vercel.app` and `vercel.app` but it didn't work, it doesn't even let me log in (I'm logging in with Google).

Do you guys know why?

**Reply by berik** [reply]
Hey @chhuang ,I had some issues with `.vercel.app` too so I've bought the custom domain and worked correctly. 
This thread helped to fix multitenant issue: https://github.com/vercel/platforms/issues/69

Hope it helps

**Reply by tomekit** [reply]
Using this approach I've got duplicate cookies, one on: `.example.com` (new) and 2nd one on the `subdomain.example.com` (original).
The issue is that I can't logout user, as only: `.example.com` cookie is being removed, where as: `subdomain.example.com` persists, so next time when user visits page they're authorized.

Also described here:
https://github.com/orgs/supabase/discussions/3198#discussioncomment-5748383

**Reply by lucasishuman** [reply]
I am following [the technique described above](https://github.com/orgs/supabase/discussions/5742#discussioncomment-4050444) in my Next.js project - I have one 'auth' subdomain that is responsible for handling sign in + setting top-level domain cookies and sign out by deleting them. Then there are several other subdomains that all require a user to be signed in.

I also noticed that the subdomains still automatically set cookies on the subdomain even though I was specifying the session to use manually.

My workaround for this is to manually strip out the auth token cookie on all my subdomains in my Next.js middleware responses (I was already ready using middleware to make sure users are signed in for all of my subdomains except the one responsible for 'auth'), like...

```js
export async function middleware(req: NextRequest) {
  // We need to create a response and hand it to the supabase client to be able to modify the response headers.
  const res = NextResponse.next();

  // Create authenticated Supabase Client.
  const supabaseClient = createMiddlewareSupabaseClient({ req, res });

  // Check if we have a session
  const {
    data: { session },
  } = await supabaseClient.auth.getSession();

  // Check auth condition
  if (session?.user) {
    // Authentication successful, forward request to protected route.
    // NOTE: deleting this subdomain-specific cookie is necessary so make the top-level cookies work
    res.cookies.delete('supabase-auth-token'); // <-- THIS IS THE IMPORTANT BIT
    return res;
  }

  const signInUrl = '[my sign in url]';
  return NextResponse.redirect(signInUrl);
}
```

I wish this wasn't necessary and maybe there's a better way to do it... 🤷🏼‍♂️

Hopefully Supabase can implement a better way to handle auth + users + sessions across subdomains in the future 🤞🏻

**Reply by JorensM** [reply]
When I try this and inspect my request in devtools, I don't see the appropriate headers being attached to the request, nor the cookie.

**Comment by Necmttn** [comment]
A small bug fix for nextjs app dir users;
https://github.com/supabase/auth-helpers/pull/604 
which enables to use subdomain auth

**Reply by kevin-eberhardt** [reply]
Is this solution also available for supabase/ssr, since auth-helpers is supposed to be replaced by it?

**Comment by thedevdavid** [comment]
Is there anything about how this works with supabase/ssr?

**Reply by uncvrd** [reply]
I'm just getting my feet wet with Supabase Auth but I found out you can set a `.yourdomain.com` cookie doing the following:

```
import { createBrowserClient } from "@supabase/ssr"

const supabase = createBrowserClient<Database>(env.NEXT_PUBLIC_SUPABASE_PROJECT_URL, env.NEXT_PUBLIC_SUPABASE_ANON_KEY, {
        cookies: {},
        cookieOptions: {
            domain: `yourdomain.com`,
        },
})
```

Typescript will give you a fuss unless you add that cookies object for some reason but this works so far...

Maybe someone else can add anything else that may be needed?

**Comment by Jaaneek** [comment]
Any updates on this?

**Comment by scrabase** [comment]
Any simple way to do this with nextjs 14?

**Comment by GabrielFabian** [comment]
Has anyone been able to redirect their users from their main domain (example.com) to their sub-domain (app.example.com) where `app.example.com` is a completely different project within vercel while remaining authenticated? I'm handling auth via middleware the [way the docs suggest ](https://supabase.com/docs/guides/auth/server-side/nextjs). 

Predictably, since the project that hosts `app.exaple.com` is on a different server, I can't pass the sessions around as I'd like.

**Reply by GabrielFabian** [reply]
I was able to figure this out, here's what I did.

- server action handles form submit on project w/ apex domain
- set `sb-auth` cookie 
- ensure cookie options object is configured correctly


```//serverActions.ts
export const signInWithPassword = async (formData: FormData) => {
	const email = formData.get("email") as string;
	const password = formData.get("password") as string;
	const supabase = createClient();

	const { data, error } = await supabase.auth.signInWithPassword({
		email,
		password,
	});

	if (error) {
		return redirect("/auth/signin?auth=failed");
	}

	if (process.env.NODE_ENV === "production") {
		cookies().set("sb-auth", JSON.stringify(data.session), { // cookie options
			httpOnly: true,
			secure: process.env.NODE_ENV === "production",
			sameSite: "lax",
			path: "/",
			domain: ".example.com",
			expires: new Date(data.session?.expires_at?.toString() || 0),
		});
		return redirect("https://app.example.com/admin");
	}

	return redirect("/admin"); // on localhost, don't worry about cookies
};
```

- in subdomain project middleware, detect cookie assigned from apex domain
- set session w/ cookie value

```
//middleware.ts
if (request.cookies.get("sb-auth")) {
		const session: Session = JSON.parse(request.cookies.get("sb-auth")!.value!);
		const response = await supabase.auth.setSession(session);
}
```

**Gotcha**: assure that on log out, you're removing `sb-auth` cookie and whatever auth cookies supabase sets.

**Comment by Rhys97W** [comment]
Anyone found a solution to this? I am trying to replicate the Vercel Platforms Starter Kit but cannot get cross-domain auth working with Supabase at all

**Reply by ramezj** [reply]
Any updates on this?

**Reply by Shafnaa** [reply]
So, basically the `createServerClient` from `@supabase/ssr` can accept cookieOptions to configure how the cookies behave in the browser. I follow this comment for the configuration, and it works for me:
[https://github.com/orgs/supabase/discussions/3198#discussioncomment-5550787](https://github.com/orgs/supabase/discussions/3198#discussioncomment-5550787)

```
createServerClient(
    process.env.NEXT_PUBLIC_SUPABASE_URL!,
    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
    {
      cookieOptions: {
        domain: process.env.NODE_ENV === "production"
            ? ".example.com"
            : "localhost",
        maxAge: 100000000,
        path: "/",
        sameSite: "lax",
        // secure: process.env.NEXT_PUBLIC_AUTH_DOMAIN.startsWith('https://'),
      },
      cookies: {
        ...
      },
    }
  );
```

If you follow the guide on the docs here [https://supabase.com/docs/guides/auth/server-side/nextjs](https://supabase.com/docs/guides/auth/server-side/nextjs). You should only need to configure 3 files, `utils/supabase/client`, `utils/supabase/server`, and `utils/supabase/middleware`. (at first I only add the cookieOptions on the `utils/supabase/server` file and when I logout using the client side. Suddenly the www.example.com cookies appear and give me some error. So, I guess you need to configure all `createServerClient` and `createServerBrowser` from `@supabase/ssr`)

**Reply by UsamaKashif** [reply]
I did the same thing but it does not work.
could share how you setup middleware.ts file with the updateSession function??

- sign in using app.localhost:3000
- it creates a cookie
- but when I visit demo.localhost:3000
- cookie is not found
- hence it redirect back to app.localhost:300

**Reply by Shafnaa** [reply]
So I'm using `@supabase/ssr@0.3.0` so it might look little bit different from the template on the docs but it's basically the same

```
import { createServerClient, type CookieOptions } from "@supabase/ssr";
import { NextResponse, type NextRequest } from "next/server";

export async function updateSession(request: NextRequest) {
  let response = NextResponse.next({
    request: {
      headers: request.headers,
    },
  });

  const supabase = createServerClient(
    process.env.NEXT_PUBLIC_SUPABASE_URL!,
    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
    {
      cookieOptions: {
        domain:
          process.env.NODE_ENV === "production"
            ? ".example.com"
            : ".localhost",
      },
      cookies: {
        get(name: string) {
          return request.cookies.get(name)?.value;
        },
        set(name: string, value: string, options: CookieOptions) {
          request.cookies.set({
            name,
            value,
            ...options,
          });
          response = NextResponse.next({
            request: {
              headers: request.headers,
            },
          });
          response.cookies.set({
            name,
            value,
            ...options,
          });
        },
        remove(name: string, options: CookieOptions) {
          request.cookies.set({
            name,
            value: "",
            ...options,
          });
          response = NextResponse.next({
            request: {
              headers: request.headers,
            },
          });
          response.cookies.set({
            name,
            value: "",
            ...options,
          });
        },
      },
    }
  );

  // refreshing the auth token
  const {
    data: { user },
  } = await supabase.auth.getUser();

  if (request.nextUrl.pathname.startsWith("/protected")) {
    if (!user) {
      const url = request.nextUrl.clone();
      url.pathname = "/login";

      return NextResponse.redirect(url);
    }
  }

  return response;
}
```
have you make sure the cookie domain is `.localhost` and not `app.localhost`?

**Reply by UsamaKashif** [reply]
Thanks, will give it a try
Can you also share how you setup the next js middleware file with this update session function to handle subdomains?

**Comment by hsab** [comment]
If anyone else is running into this I recommend having a look at https://github.com/supabase/supabase/issues/27283#issuecomment-2170478242

**Comment by thejufo** [comment]
Yes it's possible, please see my answer in this issue  -> [answer](https://github.com/supabase/supabase/issues/473#issuecomment-2543434925)

**Comment by DanielSoCra** [comment]
Has anyone found a solution for CSR React apps not build with Next but Vite for example?

**Comment by Towerful** [comment]
I know this is old, and supabase has done some awesome development over the past 2 years.  
But I also see some recent comments, and this was one of the hits I found when googling this exact issue.  

The solution is to use your own Auth storage provider in the supabase client:  

```ts
import {createClient} from '@supabase/supabase-js'
import {Database} from './generated-types.ts'
const supabase = createClient<Database>(supabaseUrl, supbaseKey, {
  auth: {
    storage: /*Your Custom Storage*/
  }
}
```
The `localStorageAdapter` & `memoryLocalStorageAdapter` (essentially the default and the fallback) are shown here:  
https://github.com/supabase/auth-js/blob/master/src/lib/local-storage.ts

the storage type can be written as follows:  
```ts
import type { SupabaseAuthClientOptions } from '@supabase/supabase-js/dist/module/lib/types'
type MyStorageProvider = Exclude<SupabaseAuthClientOptions['storage'], undefined> 
const useCookieAuthStorageProvider = () : MyStorageProvider => {
  return {
    getItem: (key) => {
      // get an item from auth storage by key
    },
    setItem: (key, value) => {
      // set an item in auth storage by key & value
    },
    removeItem: (key) => {
      // remove an item in auth storage by key & value
    }
  }
}
```

I feel like this is the correct way to do it.  
This allows you to write a cookie storage method that sets a parent domain, allowing access across all subdomains. I'm not massively convinced by the security issues of cookies vs localStorage in dedicated environments.  
But if you ARE on shared hosting under a subdomain that you do not ABSOLUTELY CONTROL, then cookies may be stolen by another subdomain (ie `my-app.example.com`  auth cookies with a parent domain of `example.com` - being stolen by `malicious.example.com`).  
So make sure your attack vectors are covered.

---

As an alternative:   

```ts
import {createClient} from '@supabase/supabase-js'
import {Database} from './generated-types.ts'
const supabase = createClient<Database>(supabaseUrl, supbaseKey, {
  accessToken: async () => {
    // get access token  
  }
  
}
```
Is a way to provide a 3rd party access token, which gets called ANY time an access token is required.  
So as long as that function returns the most recent/valid access token, it doesn't matter where it comes from.  
Just be aware that this essentially disables supabase/auth-js entirely. so you are then FULLY incharge of authenticating a user (ie using Keycloak or whatever via something like authts/oidc-client-ts  
I mention this, because using Authentik and authts/oidc-client-ts seems to solve my issue. and the `accessToken` config property doesn't really seem to be documented (although, I find the source code fairly easy to follow!)

**Reply by zackjh67** [reply]
This is definitively the best answer

**Reply by sourman** [reply]
I do not understand, why all this complexity. is not setting the cookie to be shared across subdomain sufficient? And what about @GabrielFabian 's answer? it seems if you are worried that someone can host malicous.example.com and get the user to go to it (thereby leaking the cookies) with his technique the cookies can be encrypted and decrypted per subdomain

**Comment by ricardo17coelho** [comment]
i would like to see an official example how to archieve this.
Having multiple apps running on subdomains is a common use case.
Ex:
- app.domain.com
- auth.domain.com
- admin.domain.com
- ....

**Reply by lucasishuman** [reply]
Agreed! I've rolled my own for exactly the case you outlined, but an officially recommended / supported approach and guide would be nice.

**Reply by Towerful** [reply]
I've actually gone down the route of keycloak, with oidc auth on a kubernetes gateway (envoy).  

I create an RSA256 private key, import that into keycloak and made it the only signing cert.  
I also make a JWK/S from the RSA private key and provide it to postgrest/realtime/storage with the idea of completely bypassing supabase/auth (they all support JWKS now!)

I've done some manual testing, and I can issue correct looking/validating supabase JWTs from keycloak.  
I've also done some separate testing with supabase and JWKs created from the signing key.  
Now I just need to make sure the JWT issued by keycloak actually works with supabase services, and that everything connects up nicely.

All self hosted. I don't think any of this is possible with the hosted service, other than through social logins or through Enterprise level SAML

**Comment by takeItIzzy** [comment]
If all the above attempts fail, consider deploying to a test environment first, as the issue may only affect localhost while working fine online. This could be due to limitations with setting domain on localhost, though it's unclear if this is caused by Supabase. There's a relevant discussion about this behavior in the HTTPie CLI issue tracker: https://github.com/httpie/cli/issues/602

For local development, you can modify your hosts file to point 127.0.0.1 to a test domain like yourdomain.test. Then update all instances of localhost in your codebase to use this domain name, and add it to your Supabase URL Configuration.

**Comment by sourman** [comment]
I spent three days going off different threads in this convo, it turns out the latest supabse auth supports this out of the box. I made a repo and a video showing how to make this work here https://github.com/sourman/example-supabase-auth-across-subdomains

This setup mimics the real setup with real urls so you know that it will work in prod

**Reply by pourya7** [reply]
Thanks @sourman 

The `cookieOptions` did the work for us when using `createServerClient` but using `createBrowserClient` still lacks the shared cookies between subdomains (even when we add `cookieOptions` to it)!

In your example, all your pages and components use the server side Supabase client. Have you tried it with the Supabase browser client?

**Reply by sourman** [reply]
yes it works on both the server and client side, BUT you have to set the cookies to be httpOnly=false. The supabase browser client does not work unless you do this, which is fine in most cases

basically expose the cookies to browser javascript and hope for the best :) if you are concerned about the token leaking, then I guess you could create server actions and pass only the things you need from authentication state back to the user and vise versa

I am keeping my cookies httpOnly and creating server actions to let client components do privileged actions on behalf of the user. This is probably overkill for my application as it is not a high security application

**Reply by marclelamy** [reply]
I tried to implement it locally before testing on throw away domain but I couldn't make it work on localhost. What should I replace the domain by?
```typescript
cookieOptions: { // 👈 Add this
                    domain: '.localhost', // Allow localhost and all subdomains
                    secure: false,
                    sameSite: 'lax',
                    path: '/',
                    maxAge: 60 * 60 * 24 * 30,
                    httpOnly: true,
                },
```

**Reply by sourman** [reply]
The whole setup is in my example repo. You will need to run a reverse proxy which is also detailed in my example repo.

localhost is not going to work because you will have different port numbers for each node server hosting each subdomain. The browser is not going to allow cookies sharing unless you setup the local proxy that will pipe the localhost to some fake domain name that you can then visit in the browser

**Comment by hnykda** [comment]
I don't know what has changed, but in my case, this worked "out of the box", without any "proxy" and so on. What I mean:

1. I followed https://supabase.com/docs/guides/auth/server-side/nextjs
2. I pass this for `cookieOptions`:

```typescript
const authDomain = process.env.NEXT_PUBLIC_AUTH_DOMAIN ?? undefined
export function cookiesFactory() {
  return {
    name: `sb-${authDomain}-auth-token`,
    domain: authDomain,
    path: "/",
    maxAge: 60 * 60 * 24 * 365, // 1 year
    sameSite: "lax" as const,
  };
}


// browser.ts
import { createBrowserClient } from "@supabase/ssr";

import { Database } from "../types.gen";
import { supabaseKey, supabaseUrl } from "../../constants";
import { cookiesFactory } from "../../cookieFactory";

export function createClient() {
  return createBrowserClient<Database>(supabaseUrl, supabaseKey, {
    cookieOptions: cookiesFactory(),
  });
}

// similarly for server client as well as middleware client
```
3. I use the same in two separated nextjs projects (literally importing it from the shared TS lib), run `pnpm run dev` where one is on localhost:3000 and other on localhost:3001. The `NEXT_PUBLIC_AUTH_DOMAIN` is set to the shared root domain, e.g. `example.com` for `a.example.com` and `b.example.com`. **It is undefined on localhost during dev**.
4. logging in on localhost:3001/login and then going in the same browser to localhost:3000, I am logged in and vice versa. Logging out on one also logs me out on the other.
5. works on our custom domain `a.example.com` and `.b.example.com` the same as in dev 🤷 

I admit I am not sure if all of this is needed, I am too lazy to try right now.

**Reply by NickStull** [reply]
Sorry for bugging you. But I wanted to read more in to the cookieOptions param you are passing in to createBrowserClient but I cannot find anywhere in the docs that references this as an option. Could you help direct me?

**Reply by hnykda** [reply]
Hey. I have no idea where it's documented. I found it based on the messages above, have no idea how it works.

**Reply by NickStull** [reply]
That's actually insane 🤣 appreciate the response. I'm like 90% sure it's not anywhere so how tf did the people above figure out that's an option that can be passed in. Wild.

**Comment by robbiecren07** [comment]
For anyone that is still having issues or is trying this on `"@supabase/ssr": "^0.6.1",` and `"@supabase/supabase-js": "^2.50.0",` - I had to put the cookies options within the `setAll()` like:

```jsx
setAll(cookiesToSet) {
    cookiesToSet.forEach(({ name, value }) => request.cookies.set(name, value))
    supabaseResponse = NextResponse.next({
      request,
    })
    cookiesToSet.forEach(({ name, value, options }) => {
      // Override cookie options to set domain for cross-subdomain sharing
      const cookieOptions = {
        ...options,
        // Only set domain for production/preview, not localhost
        ...(process.env.NODE_ENV === 'production' && { domain: `.${rootDomain}` }),
        secure: process.env.NODE_ENV === 'production',
        sameSite: 'lax' as const,
        path: '/',
      }
      supabaseResponse.cookies.set(name, value, cookieOptions)
    })
  },
```
        
 I also applied this to my `server.ts` as well. This works so far on vercel preview deployments, but for localhost is does not. If you try to do `domain: '.localhost'`, supabase is not able to create the cookies and you can't login on localhost, this it why I did the conditional domain logic.
 
 For my use case, im doing a web app with multi-tenant subdomains, so user logs in at `domain.com/login` and works in their dashboard at `domain.com/dashboard`, but their public pages are at `their-subdomain.domain.com` so when they navigate to their subdomain, they are still logged in.
